app/javascript/dashboard/components-next/filter/ConversationFilter.vue

```diff
@@ -0,0 +1,175 @@
+<script setup>
+import { useTemplateRef, onBeforeUnmount, onMounted, computed, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useTrack } from 'dashboard/composables';
+import { useStore } from 'dashboard/composables/store';
+import { CONVERSATION_EVENTS } from 'dashboard/helper/AnalyticsHelper/events';
+
+import Button from 'next/button/Button.vue';
+import ConditionRow from './ConditionRow.vue';
+import Dialog from 'next/dialog/Dialog.vue';
+
+const props = defineProps({
+  isFolderView: {
+    type: Boolean,
+    default: false,
+  },
+  folderName: {
+    type: String,
+    default: '',
+  },
+});
+
+const emit = defineEmits(['applyFilter', 'updateFolder', 'close']);
+const filters = defineModel({
+  type: Array,
+  default: [],
+});
+const folderNameLocal = ref(props.folderName);
+
+const { t } = useI18n();
+const store = useStore();
+
+const resetFilter = () => {
+  filters.value = [
+    {
+      attribute_key: 'status',
+      filter_operator: 'equal_to',
+      values: '',
+      query_operator: 'and',
+    },
+  ];
+};
+
+const removeFilter = index => {
+  if (filters.value.length === 1) {
+    resetFilter();
+  } else {
+    filters.value.splice(index, 1);
+  }
+};
+
+const addFilter = () => {
+  filters.value.push({
+    attribute_key: 'status',
+    filter_operator: 'equal_to',
+    values: '',
+    query_operator: 'and',
+  });
+};
+
+const conditionsRef = useTemplateRef('conditionsRef');
+const dialogRef = useTemplateRef('dialogRef');
+
+function updateSavedCustomViews() {
+  const isValid = conditionsRef.value
+    .map(condition => condition.validate())
+    .every(Boolean);
+  if (!isValid) return;
+
+  emit('updateFolder', filters.value, folderNameLocal.value);
+}
+
+function validateAndSubmit() {
+  const isValid = conditionsRef.value
+    .map(condition => condition.validate())
+    .every(Boolean);
+  if (!isValid) return;
+
+  store.dispatch(
+    'setConversationFilters',
+    JSON.parse(JSON.stringify(filters.value))
+  );
+  emit('applyFilter', filters.value);
+  useTrack(CONVERSATION_EVENTS.APPLY_FILTER, {
+    applied_filters: filters.value.map(filter => ({
+      key: filter.attribute_key,
+      operator: filter.filter_operator,
+      query_operator: filter.query_operator,
+    })),
+  });
+}
+
+const closeModal = () => {
+  dialogRef.value.close();
+  emit('close');
+};
+
+const filterModalHeaderTitle = computed(() => {
+  return !props.isFolderView
+    ? t('FILTER.TITLE')
+    : t('FILTER.EDIT_CUSTOM_FILTER');
+});
+
+onMounted(() => dialogRef.value.open());
```

Comment:
Could you help me understand why we need to open the dialog on mounted?

Reply:
The default dialog state is closed. So we manually have to open it. It's fixed in the popover PR. There we don't need this